internal/server/data/access_key.go

```diff
@@ -83,19 +83,20 @@ func CreateAccessKey(db *gorm.DB, accessKey *models.AccessKey) (body string, err
 	return fmt.Sprintf("%s.%s", accessKey.KeyID, accessKey.Secret), nil
 }
 
-func SaveAccessKey(db *gorm.DB, key *models.AccessKey) error {
+func SaveAccessKey(db GormTxn, key *models.AccessKey) error {
 	if key.Secret != "" {
 		key.SecretChecksum = secretChecksum(key.Secret)
 	}
 
 	return save(db, key)
 }
 
-func ListAccessKeys(db *gorm.DB, p *models.Pagination, selectors ...SelectorFunc) ([]models.AccessKey, error) {
+func ListAccessKeys(db GormTxn, p *models.Pagination, selectors ...SelectorFunc) ([]models.AccessKey, error) {
 	return list[models.AccessKey](db, p, selectors...)
 }
 
-func GetAccessKey(db *gorm.DB, selectors ...SelectorFunc) (*models.AccessKey, error) {
+func GetAccessKey(tx GormTxn, selectors ...SelectorFunc) (*models.AccessKey, error) {
+	db := tx.GormDB()
```

Comment:
Anything that builds a query itself using gorm methods will need to call `tx.GormDB`. Mostly that is our helpers like `add`, `save`, `get`, but also a few others that do more interesting queries.

This will give us an easy way of finding all the queries that need to be replaced with sql.

Reply:
I found this a little confusing too when reading through the PR. Maybe a note around `GormDB()` which explains when to use it?

Reply:
We should be removing `GormDB` very soon, but I'll add a comment for the interim.